packages/db/src/search/GranuleSearch.ts

```diff
@@ -116,9 +119,30 @@ export class GranuleSearch extends BaseSearch {
    * @param pgRecords - postgres records returned from query
    * @returns translated api records
    */
-  protected translatePostgresRecordsToApiRecords(pgRecords: GranuleRecord[])
-    : Partial<ApiGranuleRecord>[] {
+  protected async translatePostgresRecordsToApiRecords(pgRecords: GranuleRecord[], knex: Knex)
+    : Promise<Partial<ApiGranuleRecord>[]> {
     log.debug(`translatePostgresRecordsToApiRecords number of records ${pgRecords.length} `);
+
+    const cumulusIds = pgRecords.map((record) => record.cumulus_id);
+
+    //get Files
+    const fileModel = new FilePgModel();
+    const files = await fileModel.searchByGranuleCumulusIds(knex, cumulusIds);
+    const fileMapping: { [key: number]: PostgresFileRecord[] } = {};
+    files.forEach((file) => {
+      if (!(file.granule_cumulus_id in fileMapping)) {
+        fileMapping[file.granule_cumulus_id] = [];
+      }
+      fileMapping[file.granule_cumulus_id].push(file);
+    });
+
+    //get Executions
+    const executionUrls = await getExecutionInfoByGranuleCumulusIds({
```

Comment:
I need to go back to the drawing board on this, I didn't think properly about granule/execution association, so this bit is not working right

Reply:
👍🏻  I'll hold off on digging into the units until you've had a chance to look into it.

Reply:
this is ready to go now